app/domain/batch_process.rb

```diff
@@ -0,0 +1,17 @@
+class BatchProcess
+  attr_reader :output_documents
+
+  def initialize(output_documents)
+    @output_documents = output_documents
+  end
+
+  def csv
+    [headers].concat(output_documents.map(&:csv)).join
+  end
+
+  private
+
+  def headers
+    OutputDocument::CSVRenderer::FIELDS.join('|') + "\n"
```

Comment:
This has potential to catch is out. We're trusting that fields won't need escaping can just be concatenated. Why not use http://ruby-doc.org/stdlib-2.2.1/libdoc/csv/rdoc/index.html ?